monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -41,81 +32,168 @@ function getObjectFromRegistryByRef(refString, registry) {
   return resolveObjectPath(refArray, registry);
 }
 
-function getFullDefinitionByKey(refString, registry, itemKey) {
-  let fullArray = getFullDefinitionsFromRegistry(refString, registry);
-  return fullArray.filter(e => (e.enum[0] === itemKey))[0];
-}
-
-function setPaneInfo(refString, registry, itemKey, setPaneInfoFnc) {
-  let definitionObj = getFullDefinitionByKey(refString, registry, itemKey);
-  setPaneInfoFnc({title: definitionObj.title, content: definitionObj.info, link: definitionObj.link});
-}
-
 function getDefaultPaneParams(refString, registry) {
   let configSection = getObjectFromRegistryByRef(refString, registry);
   return ({title: configSection.title, content: configSection.description});
 }
 
-function AdvancedMultiSelect(props) {
-  const [masterCheckbox, setMasterCheckbox] = useState(true);
-  const {
-    schema,
-    id,
-    options,
-    value,
-    required,
-    disabled,
-    readonly,
-    multiple,
-    autofocus,
-    onChange,
-    registry
-  } = props;
-  const {enumOptions} = options;
-  const [infoPaneParams, setInfoPaneParams] = useState(getDefaultPaneParams(schema.items.$ref, registry));
-  getDefaultPaneParams(schema.items.$ref, registry);
-  const selectValue = cloneDeep(value);
-  return (
-    <div className={'advanced-multi-select'}>
-      <Card.Header>
-        <Button key={`${props.schema.title}-button`} value={value}
+class AdvancedMultiSelect extends React.Component {
```

Comment:
Why did you refactor to class component? Functional components usually requires less boilerplate and is more readable

Reply:
It simplified some of the callbacks and reduced the number of things that were being passed back and forth between the parent and child components.

Reply:
It simplified callbacks not because it's a class component, but because functions are now inside the component and access component state directly. The drawback is that now it's less testable, but that's ok.

Reply:
I'm not thrilled with the change, but I think it keeps the child components simpler and the  code is overall more readable. As the component evolves, we can look for ways to further simplify it and maybe switch back to a functional component.